### PR TITLE
Fix/deal duration

### PR DESCRIPTION
**Pull request by sshaman1101**
This PR implements deal duration.

Changed:
- Slot now have duration (in seconds);
- When the Node is opening a deal it will add a correct order duration to a deal (and writing it to the blockchain).

Added: 
- the Order and the Slot now accept human-readable values for byte fields (RAM, storage, net in-out)

Fixed:
- accidentally fix the orderHandler processing interruption when the Node cannot resolve Hub IP via the Locator;

### CODE REVIEW

File: cmd/cli/task_config/slot.go [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
-
 type RatingConfig struct {
 	Buyer    int64 `yaml:"buyer" required:"true"`
 	Supplier int64 `yaml:"supplier" required:"true"`
 }
 
 type ResourcesConfig struct {
 	Cpu        uint64             `yaml:"cpu_cores" required:"true"`
-	Ram        uint64             `yaml:"ram_bytes" required:"true"`
+	Ram        string             `yaml:"ram_bytes" required:"true"`
```
Comment (antmat) [inline cmd/cli/task_config/slot.go]: Maybe next time better to make separate PR?

File: insonmnia/marketplace/server_test.go [hunk 2]
```diff
@@ -258,6 +267,7 @@ func TestInMemOrderStorage_GetOrders_Count(t *testing.T) {
 			Price:     "1",
 			OrderType: pb.OrderType_BID,
 			Slot: &pb.Slot{
+				Duration:  uint64(structs.MinSlotDuration.Seconds()),
```
Comment (antmat) [inline insonmnia/marketplace/server_test.go]: How many such slots do you have?
I wonder if you changed those with some kind of sed or did it by hand :)

File: insonmnia/node/market.go [hunk 3]
```diff
@@ -322,12 +323,19 @@ func (m *marketAPI) CreateOrder(ctx context.Context, req *pb.Order) (*pb.Order,
 		return nil, errNotAnBidOrder
 	}
 
+	if time.Duration(req.GetSlot().GetDuration())*time.Second < structs.MinSlotDuration {
```
Comment (antmat) [inline insonmnia/node/market.go]: why don't you use structs.Slot?

File: insonmnia/marketplace/server_test.go (hunk 2 above)
Comment (sshaman1101) [reply insonmnia/marketplace/server_test.go]: It made by hand, of course!

File: insonmnia/node/market.go (hunk 3 above)
Comment (sshaman1101) [reply insonmnia/node/market.go]: Hmm, seems reasonable, will do

**Comment by 3Hren** [comment]
Hub part: LGTM for now (after private discussion).

**Comment by 3Hren** [comment]
Voted for default slot for tests. The boilerplate gauge hit its historical value.

**Comment by sshaman1101** [comment]
Make boilerplate ~~great again~~ reduction. @3Hren @antmat PTAL!
